Enable PSA APIs on Pelion boards

alzix opened the pull request:
### Description

Add PSA label - enabling PSA APIs support on Pelion boards
- K66F
- K66F, NUCLEO_F429ZI
- NUCLEO_F411RE
- UBLOX_EVK_ODIN_W2

Depends on: #9362  

### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [x] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

<!-- 
    Optional
    Request additional reviewers with @username
-->
@orenc17 @shelib01

Commit:
Enable PSA APIs on Pelion boards

Add PSA label - enabling PSA APIs support on Pelion boards
- K66F
- K66F, NUCLEO_F429ZI
- NUCLEO_F411RE
- UBLOX_EVK_ODIN_W2

ciarmcom commented:
@alzix, thank you for your changes.
@shelib01 @orenc17@ARMmbed/mbed-os-maintainers please review.

orenc17 reviewed (changes requested):
Add configurations to KVStore and PSA-Crypto

alzix commented:
cc @davidsaada @dannybenor

Commit:
Add FLASHAIP component

Commit:
Enable entropy injection on PSA targets

enable entropy injection on:
- K64F
- K66F
- NUCLEO_F411RE
- NUCLEO_F429ZI
- UBLOX_EVK_ODIN_W2

orenc17 commented on targets/targets.json:
```diff
@@ -1451,7 +1451,14 @@
+            "CPU_MK64FN1M0VMD12",
+            "FSL_RTOS_MBED",
+            "MBEDTLS_PSA_CRYPTO_C",
+            "MBEDTLS_ENTROPY_NV_SEED",
```
This device has TRNG enabling nvseed is redundant

alzix replied:
Pelion require injection API regardless TRNG

orenc17 commented on targets/targets.json:
```diff
@@ -1735,10 +1742,18 @@
+            "CPU_MK66FN2M0VMD18", 
+            "FSL_RTOS_MBED", 
+            "MBEDTLS_PSA_CRYPTO_C",
+            "MBEDTLS_ENTROPY_NV_SEED",
```
This device has TRNG enabling nvseed is redundant

alzix replied:
Pelion require injection API regardless TRNG

orenc17 commented on targets/targets.json:
```diff
@@ -2655,11 +2678,17 @@
-            "USBHOST_OTHER"
+            "USBHOST_OTHER",
+            "MBEDTLS_PSA_CRYPTO_C",
+            "MBEDTLS_ENTROPY_NV_SEED",
```
This device has TRNG enabling nvseed is redundant

alzix replied:
Pelion require injection API regardless TRNG

orenc17 commented on targets/targets.json:
```diff
@@ -4132,6 +4161,14 @@
+        "components_add": ["FLASHIAP"],
+        "macros_add": [
+            "MBEDTLS_PSA_CRYPTO_C",
+            "MBEDTLS_ENTROPY_NV_SEED",
```
This device has TRNG enabling nvseed is redundant

alzix replied:
Pelion require injection API regardless TRNG

Commit:
Set "TDB_INTERNAL" default storage for Pelion

NUCLEO_F411RE
NUCLEO_F429ZI
UBLOX_EVK_ODIN_W2

Commit:
fix its_reset test

0xc0170 commented:
I marked this for 5.11.2 (as I understood from communication, this is important)

0xc0170 commented:
Please review

dannybenor commented:
@theamirocohen please add the PRs this PR depends on

theamirocohen commented:
Depends on #9241 and #9244

0xc0170 commented:
> #9244 

While we wait for the last dependancy, this can be reviewed !

cmonr reviewed (approved):
LGTM

cmonr commented:
CI started

dannybenor commented:
It needs 9244

0xc0170 commented:
> It needs #9244

Integrated, CI restarted now

cmonr commented:
Looks like `jenkins-ci/greentea-test` is about to fail.

`jenkins-ci/cloud-client-test` failure appears to be an issue with connecting to DUTs. First failure of its kind today, so considering it an isolated case.

Log files for `jenkins-ci/greentea-test` will need to be reviewed before CI is restarted.

dannybenor commented:
@alzix can you have a look at mbedtls_entropy_self_test?

teetak01 commented:
@cmonr the Client failures looks like potentially real failures due to this PR (SecureConnectionFailed).

0xc0170 commented:
> @cmonr cmonr removed the release-version: 5.11.2  label a day ago

@cmonr why was the label removed? @dannybenor please let us know how to proceed

cmonr commented:
@0xc0170 I originally removed it because last night it was looking like it wouldn't make it in time. 
Have added it back in.

NirSonnenschein commented:
re-running CI after rebase

Commit:
Revert "Enable entropy injection on PSA targets"

mbedtls performs selftest on all entropy sources.
Enable injected source without proper entropy injection causes mbedtls
selftest to fail.

Application code can still enable entropy injection by adding the macros
to mbed_app.json/mbed_lib.json

This reverts commit 8c03e3e60d46da71708c4a0e4dacf46a0179d18d.

NirSonnenschein commented:
re-start CI

Commit:
Fix the general block device to support better low memory boards

In addition, fix prints and warnings.

dannybenor commented:
This will need https://github.com/ARMmbed/mbed-os/pull/9362 to pass (not enough ram to run the test)

alzix commented:
@dannybenor 
> This will need #9362 to pass (not enough ram to run the test)

rebased on top of `davidsaada`:`david_bd_test_low_mem`

NirSonnenschein commented:
starting CI to see if the issues are resolved after rebase
